blueocean-dashboard/src/main/js/redux/actions.js

```diff
@@ -383,11 +384,14 @@ export const actions = {
                         // run. No need to create another i.e. ignore this event.
                         return;
                     }
+                    if (parseInt(run.id, 10) > nextId) { // need the estimated next id
+                        nextId = parseInt(run.id, 10);
+                    }
                 }
```

Comment:
The issue is now we fetch a run when the RunDetails page is loaded, before we were fetching all runs and passing a run object along. If an item is actually queued, without a runId, it results in an invalid route. We need a `runId` for the route to work properly, and then fetch-or-not based on what we see in the cache. There is still a problem that refreshing the page would not work because you can't fetch a queued item by runId (even though it shows up in the activity list with an estimated run id, calculated similarly to the code here).